zerver/migrations/0482_automatically_follow_unmute_topics_policy_defaults.py

```diff
@@ -0,0 +1,86 @@
+# Generated by Django 4.2.6 on 2023-10-20 23:43
+
+from django.db import migrations, models
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+from django.db.models import Max
+
+AUTOMATICALLY_CHANGE_VISIBILITY_POLICY_ON_SEND = 2
+AUTOMATICALLY_CHANGE_VISIBILITY_POLICY_ON_INITIATION = 3
+
+
+def set_default_user_topic_policies(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    RealmUserDefault = apps.get_model("zerver", "RealmUserDefault")
+    UserProfile = apps.get_model("zerver", "UserProfile")
+
+    max_id_realm_user_default = RealmUserDefault.objects.aggregate(Max("id", default=0))["id__max"]
+    max_id_user_profile = UserProfile.objects.aggregate(Max("id", default=0))["id__max"]
+
+    BATCH_SIZE = 5000
+
+    lower_bound = 0
+    while lower_bound < max_id_realm_user_default:
+        RealmUserDefault.objects.filter(
+            id__gt=lower_bound, id__lte=lower_bound + BATCH_SIZE
+        ).update(
+            automatically_follow_topics_policy=AUTOMATICALLY_CHANGE_VISIBILITY_POLICY_ON_INITIATION,
+            automatically_unmute_topics_in_muted_streams_policy=AUTOMATICALLY_CHANGE_VISIBILITY_POLICY_ON_SEND,
+        )
+        lower_bound += BATCH_SIZE
+
+    lower_bound = 0
+    while lower_bound < max_id_user_profile:
+        UserProfile.objects.filter(id__gt=lower_bound, id__lte=lower_bound + BATCH_SIZE).update(
+            automatically_follow_topics_policy=AUTOMATICALLY_CHANGE_VISIBILITY_POLICY_ON_INITIATION,
+            automatically_unmute_topics_in_muted_streams_policy=AUTOMATICALLY_CHANGE_VISIBILITY_POLICY_ON_SEND,
+        )
+        lower_bound += BATCH_SIZE
```

Comment:
With `atomic=False`, this algorithm does not guarantee that we actually update every `UserProfile` row, if new ones are created while it is running, in the very unlikely event that `max_id_user_profile` is divisible by `BATCH_SIZE`. This can be more or less safely prevented by doing a `+1000` in computing `max_id_user_profile` etc.

Reply:
> For the reason, I have mentioned above the new rows created will already have the value we want. So doesn't matter.

As I noted above, this statement isn't correct.  The new rows' value is controlled by the Django code that they're running.  But in this case, since Zulip Cloud doesn't have 0477 yet, user creation will simply fail after 0482 runs until Django is restarted, since the column will be marked `not null` but Django will not be setting a value during creation, since it won't know about the column yet.

The key take-away here is that Django column defaults _are not database defaults_.